\chapter{Materials and Methods}
\section{Multiple Sequence Alignment}

Multiple sequence alignment (MSA) is an essential and well-studied fundamental problem in bioinformatics. It is used to compare and analyze the similarities between biological sequences. These sequences can be DNA, RNA, or protein sequences and can come from various species or different regions of the same genome. By aligning the sequences, it is possible to identify conserved regions that are important for the biological function or for evolution, as well as unique features that differentiate the sequences. Technically, alignments are generated and analyzed with computational algorithms. Fig.~\ref{FIG1} presents a sample (residue 1 to 60) of the alignment results of two sequences. Each row is the sequence of one of the molecules entered into the alignment. In this case, the top row is the sequence of the GSTD1 protein and the second row is the sequence of GSTE1 protein, where each vertical column represents the amino acid at a position in the sequence. Underneath the bottom row for each block is a symbolic conservation score. A (-) indicates a gap between the two sequences, this happens when the two sequences are of different length. A letter indicates that the amino acid is the same for the 2 sequences at that position. A (+) indicates the sequences have different amino acids at that position, but that the chemical properties of the different amino acids are pretty similar. When an amino acid is substituted by a very similar amino acid at a position in an alignment, we call it a conservative substitution. A blank space ( ) indicates that the amino acids are very different at that position. We call that a non-conservative substitution.
\begin{itemize}
	\item Conservative substitution: The replacement of an amino acid residue in one protein by one with similar chemical properties in another protein. For example, the replacement of serine by threonine, or leucine by isoleucine, are both considered conservative substitutions.
	\item Non-conservative substitution: The replacement of an amino acid residue in one protein by one with dissimilar chemical properties in another protein. For example, the negatively charged glutamate amino acid has quite different properties from the small glycine residue, so this is an example of a non-conservative substitution.
\end{itemize}

\begin{figure}[h!]
	\begin{verbatim}
		GSTD1  SPC-RSVIMTAKAVGVELNKKLLNLQAGEHLKPEFLKINPQHTIPTLVDNGFALWESRAI
		GSTE1  SPCVRTVKLTLKVLNLDYEYKEVNLQAGEHLSEEYVKKNPQHTVPMLDDNGTFIWDSHAI
		       SPC R+V +T K + ++   K +NLQAGEHL  E++K NPQHT+P L DNG  +W+S AI
	\end{verbatim}	
	\caption{Sample of a multiple sequence alignment.}
	\label{FIG1}
\end{figure}

In the present, we focus on the ensemble of the 25 GST sequences from a model organism. Each sequence has a different length, making the alignment process particularly useful to extract properties along the sequence at a given position. The MSA was performed using the Blast program. Through the MSA, we aim to identify regions of conservation in the 25 GST sequences, particularly for amino acids involved in the binding sites and the interface of dimerization, which are two regions of the GST proteins that are very important for the biological function (see section X). 

\section{AlphaFold}
In the present work, we used the AlphaFold2-multimer program (add ref Mirdita M, Sch√ºtze K, Moriwaki Y, Heo L, Ovchinnikov S, Steinegger M. ColabFold: Making protein folding accessible to all. Nature Methods, 2022). The program implemented in Python (Jupyter Notebook) is available online here:\\

\url{https://colab.research.google.com/github/sokrypton/ColabFold/blob/main/AlphaFold2.ipynb}\\

We used the default options, as already given in the Jupyter Notebook. At the end of the execution, the program proposes 5 different models of the 3-D structure of the protein and evaluates the predicted structures. Here, we consider only the best prediction available.

\section{Anisotropic Network Model}

The Anisotropic Network Model (ANM) is a simple yet powerful model made for Normal Mode Analysis of proteins, which has been successfully applied for deciphering the relationship between the biological function and the strucutral dynamics for many proteins\cite{ANM-COM}. In this model, the protein is represented as an elastic mass-and-spring network. In the network each node is the center of mass of the amino acid and the springs represent the interactions between the nodes. 

\begin{figure}[h!]
		%		
	\begin{minipage}{.48\linewidth}
		A)\\
		\includegraphics[width = 8cm]{figures/GSTD1_ANM-COM_Bfactors_structure.png}	
	\end{minipage}
	\begin{minipage}{.48\linewidth}
		B)\\
		\includegraphics[width = 8cm]{figures/GSTD1_ElasticNetwork.png}
	\end{minipage}
	\caption{A) Cartoon representation of the GSTD1 protein. B) ANM representation of the GSTD1 protein as an elastic mass-and-spring network.}	
	\label{FIG2}
\end{figure}

\subsection{Theory}

The potential energy function is the sum of harmonic potentials between interacting nodes:
\begin{equation}
	V=\dfrac{\gamma}{2}\sum_{|R_{ij}^0|<R_c}(R_{ij}-R_{ij}^0)^2
\end{equation}
where $R_{ij}$ is the distance between atoms $i$ and $j$ and $R_{ij}^0$ is the distance between the atoms in the reference structure, e.g., the AlphaFold structure. The summation is only performed over atoms less than a cutoff distance $R_c$, and $\gamma$ is the force constant for the elastic bond between the atoms and is the same for all atoms pairs. In the Anisotropic Network Model, the mass-weighted Hessian matrix $\mathbf{\widehat{H}}$ for a network of $N$ nodes, derived from the second derivatives of the potential energy with respect to node positions, is of the form:
\begin{equation}
	\label{mass-weighted hessian matrix}
	\hat{H}_{ij} = -\gamma\dfrac{\Gamma_{ij}}{\sqrt{M_iM_j}}\cfrac{\vv{R_{ij}}\vv{R}_ {ij}^T}{R^2_ {ij}}
\end{equation}
where $\gamma$ is the spring constant used to model interactions between nodes, $M_i$ the mass of the node $i$ and $\Gamma$ is the contact matrix. In the case $i = j$, the force matrix is computed so that the self interacting term is the response to all the applied forces:
\begin{equation}
	\label{self interacting terms}
	\hat{H}_{ii} = -\sum_{j \ne i}\hat{H}_{ij}
\end{equation}
The normal modes are obtained by diagonalization of the mass-weighted Hessian matrix:
\begin{equation}
	\label{eigen equation}
	\hat{H}\vv{e}_k = \tilde{\omega}_k^2\vv{e}_k
\end{equation}
with $3N-6$ non eigenvalues $\omega_k$ and their corresponding eigenvectors $\vv{e}_k$.\\

From the eigenvalues and eigenvectors of the mass-weighted Hessian matrix, one can compute two metrics of interest which characterize the local and non-local flexibility of a protein: the local mean square fluctuations (MSF) of a node $i$ and the non-local mean square fluctuations (MSF) of a pair node $(i,j)$ as:

\begin{equation}
	\label{thermal B-factors}
	\sigma^2(\vv{R}_i) = k_BT\sum_{k=1}^{3N-6}\frac{1}{\tilde{\omega}_k^2} \cfrac{\vert\vv{e}_{k, i}\vert^2}{M_i}
\end{equation}
\begin{equation}
	\label{metric dij}
	\sigma^2 (\vv{R}_{ij}) = k_BT\sum_{k=1}^{3N-6} \frac{1}{\omega_k^2}\left( \frac{\vv{e}_{k, j}}{\sqrt{M_j}} - \frac{\vv{e}_{k, i}}{\sqrt{M_i}} \right)^2
\end{equation}

Note that the local MSF $\sigma^2(\vv{R}_i)$ can be measured experimentally using XRD. It corresponds to the thermal B-factors corresponding to:

\begin{equation}
	B_i=\dfrac{8\pi^2}{3}\sigma^2(\vv{R}_i)
\end{equation}

When available, we compared the predicted values of the thermal B-factors with XRD data by computing the Pearson correlation coefficient as:

\begin{equation}
	\label{pearsonr}
	\rho = \cfrac{\displaystyle\sum_i (\beta_i - <\beta>_i)(B_i - <B>_i)}{\sqrt{\displaystyle\sum_i(\beta_i - <\beta>_i)^2}\sqrt{\displaystyle\sum_j(B_j - <B>_j)^2}}
\end{equation}
The values of $\rho$ are between $-1$ and $1$ and gives the linear correlation between predicted and measured B-factors. A coefficient of $1$ is associated to a perfect correlation whereas a coefficient of $0$ means that there is basically no links between prediction and experiment. 

\subsection{Parameterization}

In the ANM, two parameters are needed, \textit{i.e.} the spring constant $\gamma$ and the cutoff distance $R_c$ in order to characterize properly the local and non-local flexibility (MSF) of the proteins. To compute the contact matrix $\Gamma$ in Eq.~(\ref{mass-weighted hessian matrix}) and given two nodes $i$ and $j$, we consider $\Gamma_{ij} = 1$ if the distance $\vert\vv{R}_{ij}\vert \leq R_c$. At large distances, the interactions between the nodes are supposed to be small if not null. Therefore, $R_c$ is supposed to be here as small as possible. Moreover, the first six eigenfrequencies of the mass-weighted Hessian matrix are supposed to be zero, corresponding to the three global translations and rotations. In the case where $R_c$ is too small, the number of zero eigenfrequencies is larger than six, with no physical justifications. Therefore, the parameterization of $R_c$ is achieved with the smallest value that gives exactly $6$ zero eigenfrequencies. 

\noindent From Eq.~(\ref{mass-weighted hessian matrix}), it appears that  $\gamma$ is just a scaling factor. It can be estimated by comparing thermal B-factors computed using ANM with the ones measures by XRD  using least squared methods.

\begin{figure}[h!]
	\begin{minipage}{.48\linewidth}
		A)\\
		\includegraphics[width = 8cm]{figures/GSTD1_ANM-COM_Rc_param.jpg}
	\end{minipage}
	\begin{minipage}{.48\linewidth}
		B)\\
		\includegraphics[width = 8cm]{figures/GSTD1_ANM-COM_Bfactors.jpg}
	\end{minipage}
	
	\caption{Parametrization of the ANM applied to the GSTD1 structure.}
	\label{params}	
\end{figure}

\noindent Fig. \ref{params} shows the parameterization of the cutoff $R_c$ obtained for ANM applied to the GSTD1 protein. From panel B, we can see that for $R_c = 7.5\AA$, the eigenvalues for $k \ge 6$ are no longer zero.

\noindent From predicted and measured thermal B-factors, it apears that the optimal $\gamma$ value is 14 kcal.mol$^{-1}$.\AA$^{-2}$ from least square fit as shown in Fig. \ref{params}